### ISSUE TITLE
[Test-Proxy] Breaking Change - Move `AllowAutoRedirects` to `Transport` customization

### ISSUE THREAD

**Issue by scbedd** [Azure/azure-sdk-tools#3488]
Now that we can customize the client used on a per-recording basis, we really should move the `AllowAutoRedirects` option that is currently at the surface level of the request body to `Admin/SetRecordingOptions` into the `Transport` object.

What this means:

Current:
```
POST /Admin/SetRecordingOptions

{
   "AllowAutoRedirects": true
}
```

After

```
POST /Admin/SetRecordingOptions

{
   "Transport": {
      "AllowAutoRedirects": true
      ...other transport settings like certificate
   }
}
```

We will allow the passing of header `x-recording-id` to offer users the ability to allow redirects on a single recording, while allowing all the others to continue working without that customization.

I believe that @timovv currently uses this functionality for the js livetests. However I end up modifying this should be also pushed into `main` with the PR that updates the test-proxy version.

TODO
- [ ] Move AllowAutoRedirects into `Transport` and test
- [ ] Add new key for `Insecure` flag into `Transport`

**Comment by timovv** [comment]
> I believe that @timovv currently uses this functionality for the js livetests. However I end up modifying this should be also pushed into main with the PR that updates the test-proxy version.

That's right, JS is using this to make sure the browser tests work. Should be an easy change on the JS side, just keep me in the loop when you're looking to actually do this so we can keep everything in sync :)
